docs/user_guide/examples.md

```diff
@@ -1,14 +1,248 @@
+This is very useful, when parallelising jobs on remote clusters, since multiple branches can run simultaneously, without the need to run all branches from one task beforehand.
+
+Last, the class attribute ```reqs```, that is a {py:class}`~columnflow.tasks.framework.base.Requirements` object, does not directly have to do with the other two requirements, but is rather for analyses specific configuration.
+Strictly speaking, it is not needed for the task to run, however it offers a container for the requirements, which becomes very convenient, when for example a required task should be replaced by a customized subclass of the same task.
+Instead of having to edit the requirements in all the class methods that depends on this requirement, the ```reqs``` attribute can be simply modified with:
+```python
+MyTask.reqs.RequiredTask = ModifiedRequiredTask
+```
+Therefore, using the class attribute ```reqs```, when working with the requirements within the class is conveninet.
+
+
+![req plot](cf_plot.svg "image")
+
+### RunOnceTask
+
+The {external+law:py:class}`~law.tasks.RunOnceTask` is a special task that is used to run a task only once, which can be very useful in many situations.
+For example, when a task is required by multiple tasks or multiple branches of a workflow, but the task itself should only be executed once independent of the branches, or a task does not produce any file output but is needed to be marked as done (e.g. a task only produces text output on the CLI).
+
+This is implemented by (additionally) inheriting from {external+law:py:class}`~law.tasks.RunOnceTask`.
+Moreover, to mark the task as complete, the {external=law:py:meth}`~law.tasks.RunOnceTask.mark_complete()` method is to be added at the end of the ```run()``` method after the logic of the task is executed, or the complete```run()``` method is either to be wrapped by the {external=law:py:meth}`~law.tasks.RunOnceTask.complete_on_success` decorator.
+One must be careful with the latter, since the decorator will mark the task as complete even if it fails.
+Therefore, it is recommended to also add a check at the end and raise an error in case the task fails.
+
+```python
+import law
+
+from columnflow.tasks.framework.base import AnalysisTask
+
+class MyPrintingTask(AnalysisTask, law.tasks.RunOnceTask):
+
+    @law.tasks.RunOnceTask.complete_on_success
+    def run(self):
+        # Do something here
+        if fail:
+            raise Exception("this task just failed")
+
+    # Alternative
+    def run(self):
+        # Do something here
+        self.mark_complete()
+```
+
+### Add the task to the analysis
+
+After creating a new task, make sure to add the new task to the law config file in your analysis folder, in order to run the task.
+The new task has to be specified under the ```[outputs]``` section as follows:
+
+```ini
+[outputs]
+
+lfn_sources: wlcg_fs_desy_store
+
+cf.MyNewTask: wlcg
```

Comment:
I think you only need the `cf.` prefix if you actually implement something within the columnflow name space. In praxis, I think users wouldn't want to do that though, right?